### DISCUSSION TITLE
nl80211: Failed to create a P2P Device interface p2p-dev-wlan0

### DISCUSSION THREAD

**Discussion by EnteSuesssauer15** [RaspAP/raspap-webgui#1067] [Q&A]
![Bild_2021-11-24_201036](https://user-images.githubusercontent.com/87829672/143299754-cb1d640c-032f-41ca-97be-c6f2fc5c6cc2.png)

**Comment by billz** [comment]
Check the status of wpa_supplicant with `sudo systemctl status wpa_supplicant.service` and restart it if necessary: `sudo systemctl restart wpa_supplicant.service`. 

The big 'Re-initialize' button in your screenshot will effectively do the same thing for you.

**Reply by EnteSuesssauer15** [reply]
I pressed that button a few times and got the error but after I successfully fixed all the errors and got p2p running he found networks and I connected to mine and straight after I started the access point I couldn't connect to it and it doesn't show up in my router

**Reply by billz** [reply]
Your bash shell indicates `rfkill: WLAN soft blocked`. 
This is because you haven't set your WiFi country in raspi-config's localisation options. 
Refer to the quick start on the RaspAP website.

**Reply by EnteSuesssauer15** [reply]
I set it, then I unblocked the wlan, killed everything with wpa and then ran the Wi-Fi scan command and it worked for me but everything broke as I started the hotspot

**Reply by billz** [reply]
"everything broke" is not useful for troubleshooting.
try this https://docs.raspap.com/ap-basics/#troubleshooting

**Reply by EnteSuesssauer15** [reply]
With everything broke I ment the stuff I wrote down before "straight after I started the access point I couldn't connect to it and it doesn't show up in my router"

**Reply by billz** [reply]
use the link above and learn to capture Linux log data.

**Reply by EnteSuesssauer15** [reply]
Okay, so i just installed everything with the same errors, fixed them and ran the scan fix connected to my wifi  but i cant enable wifi client ap mode for whatever reason

![EEDB5686-0382-4D24-AC24-4D158AF2BF74](https://user-images.githubusercontent.com/87829672/143464237-ebb59ffd-03ec-4d93-9e92-4b2d41a07a52.png)

**Reply by billz** [reply]
> i cant enable wifi client ap mode for whatever reason
 
Read the docs https://docs.raspap.com/ap-basics/#wifi-client-ap-mode

This will help you develop an understanding of what you're trying to achieve.

**Reply by EnteSuesssauer15** [reply]
O pressed on save settings without changing anything and after that I was able to enable it, that's stupid
